### PR TITLE
add navigation menu

### PR DESCRIPTION
**Pull request by Zibi95**
[DEMO LINK](https://Zibi95.github.io/layout_moyo-header/)
[TEST REPORT LINK](https://Zibi95.github.io/layout_moyo-header/report/html_report/)

### CODE REVIEW

File: src/index.html [hunk 1]
```diff
@@ -1,14 +1,67 @@
...
+    <header class="header">
+      <img
+        src="./images/logo.png"
+        alt="Moyo logo"
+        width="40"
+        height="40"
+        class="header__img_logo"
+      >
+
+      <nav class="header__nav">
```
Comment (Radoslaw-Czerniawski) [inline src/index.html:28]: If you are trying to use BEM methodology here, you have to specify a separate class for a block `class="header__nav nav"`. I would personally do something like this here `class="nav"`

File: src/index.html [hunk 2]
```diff
@@ -1,14 +1,67 @@
...
+        <ul class="nav__list">
+          <li class="nav__item">
+            <a href="#apple" class="nav__item_link is-active">apple</a>
```
Comment (Radoslaw-Czerniawski) [inline src/index.html:31]: In BEM methodology every element is connected to the block. In this case your block is the nav tag. I would change the name of the classes from `nav__item_link` to `nav__link`

File: src/index.html [hunk 3]
```diff
@@ -1,14 +1,67 @@
...
+
+          <li class="nav__item">
+            <a href="#samsung" class="nav__item_link">samsung</a>
+          </li>
+
+          <li class="nav__item">
+            <a href="#smartphones" class="nav__item_link">smartphones</a>
+          </li>
+
+          <li class="nav__item" data-qa="hover">
```
Comment (Radoslaw-Czerniawski) [inline src/index.html:42]: According to the requirements data-qa attribute should be given to link tag not list item.

File: src/style.css [hunk 4]
```diff
@@ -1 +1,60 @@
...
+.header {
+  display: flex;
+  justify-content: space-between;
+  align-items: center;
+  box-shadow: 0 2px 4px rgba(0, 0, 0, 0.05);
+  padding: 0 50px;
+}
+
+.header__img_logo {
+  margin: 10px 0;
```
Comment (Radoslaw-Czerniawski) [inline src/style.css:21]: There is no need for margin here.

File: src/style.css [hunk 5]
```diff
@@ -1 +1,60 @@
...
+
+.nav__list {
+  font-size: 12px;
+  font-weight: 500;
+  text-transform: uppercase;
+  list-style: none;
+  display: flex;
+}
+
+.nav__item:nth-of-type(n + 2) {
```
Comment (Radoslaw-Czerniawski) [inline src/style.css:35]: You could also do here `.nav__item:not(:first-of-type)` but your solution still works

File: src/style.css [hunk 6]
```diff
@@ -1 +1,60 @@
...
+  margin-left: 20px;
+}
+
+.nav__item_link {
+  display: block;
```
Comment (Radoslaw-Czerniawski) [inline src/style.css:40]: No need for this.

File: src/style.css [hunk 7]
```diff
@@ -1 +1,60 @@
...
+  text-decoration: none;
+  color: black;
+}
+
+.nav__item_link:hover,
+.is-active {
+  color: var(--hover-active-color);
+}
+
+.is-active::after {
```
Comment (Radoslaw-Czerniawski) [inline src/style.css:52]: Generally you want your after element to be positioned absolute according to his relative as not to take space in the block. I would personally set first `nav__item` as position relative and set `::after` on that `nav__item` with position absolute

File: src/style.css [hunk 8]
```diff
@@ -1 +1,60 @@
...
+  content: "";
+  display: block;
```
Comment (Radoslaw-Czerniawski) [inline src/style.css:54]: This then will be redundant.

File: src/style.css [hunk 9]
```diff
@@ -1 +1,60 @@
...
+  position: relative;
+  bottom: 4px;
+  height: 4px;
+  border-radius: 2px;
+  background-color: var(--hover-active-color);
+}
```
Comment (Radoslaw-Czerniawski) [inline src/style.css:60]: What do you think about such structure?

```
.nav__item {
  height: 64px;
  position: relative;
}

.nav__item:first-of-type {
  margin-left: 20px;
}

.nav__item_link {
  height: 100%;
  display: flex;
  align-items: center;
  text-decoration: none;
  color: black;
}

.nav__item_link:hover {
  color: var(--hover-active-color);
}

.is-active {
  color: var(--hover-active-color);
}

.nav__item_link.is-active::after {
  content: "";
  position: absolute;
  left: 0;
  right: 0;
  bottom: 0;
  height: 4px;
  border-radius: 2px;
  background-color: var(--hover-active-color);
}
```

**Review by Radoslaw-Czerniawski — CHANGES_REQUESTED** [review]
Some more comments on this one. It looks fine apart from mistakes like missing `#home` link on logo. Please correct according to review and ask if you have any questions.

File: src/style.css (hunk 5 above)
Comment (Zibi95) [reply src/style.css:35]: less math less problems 😂

File: src/style.css (hunk 9 above)
Comment (Zibi95) [reply src/style.css:60]: `.nav__item {
  height: 64px;
  position: relative;
}`
height should be 60 px

`.nav__item:first-of-type {
  margin-left: 20px;
}`
I think you mean here :not(first-of-type)

Yea I think position absolute is better. I don't have to take into account this ::after height. Thanks for the insights!
I added external reset.css file as you suggested with [Meyer reset](https://meyerweb.com/eric/tools/css/reset/) but ES lint is giving me hard time with that 😂

Comment (Radoslaw-Czerniawski) [reply src/style.css:61]: - Yup now I see the design in Figma says 60px
- Also I ment something like
```
.nav__item {
  height: 64px;
  position: relative;
  margin-left: 20px;
}

.nav__item:first-of-type /* or :first-child */ {
  margin-left: 0;
}
```
My bad with the advice, reviewing css is always challenging as you are not actually seeing the results 😅
- I would import the reset pretty similarly
- also for the future if you have more than 2 css files you may create one that only imports the others then you only insert that one to the html -> however that's my personal preference

**Review by Radoslaw-Czerniawski — APPROVED** [review]
Solid work!

**Review by mkyloa — APPROVED** [review]

**Pull request closed without merging**